storage/storage.go

```diff
@@ -1068,25 +1068,32 @@ func (o *ObjectHandle) SoftDeleted() *ObjectHandle {
 	return &o2
 }
 
-// Restore will restore a soft-deleted object to full object status.
+// RestoreOptions allows you to set options when restoring an object.
+type RestoreOptions struct {
+	/// CopySourceACL indicates whether the restored object should copy the
+	// access controls of the source object. Only valid for buckets with
+	// fine-grained access. If uniform bucket-level access is enabled, setting
+	// CopySourceACL will cause an error.
+	CopySourceACL bool
+}
+
+// Restore will restore a soft-deleted object to a live object.
 // Note that you must specify a generation to use this method.
```

Comment:
We really should have samples for this feature; people are going to be so confused otherwise.